src/js/components/OptionalPortsAndServiceDiscoveryComponent.jsx

```diff
@@ -246,7 +257,11 @@ var OptionalPortsAndServiceDiscoveryComponent = React.createClass({
   },
 
   render: function () {
-    if (isTooComplexStructure(this.props.fields)) {
+    var hasPlaceIdSet = PluginComponentStore.hasComponentAtPlaceId(
+      PluginMountPoints.OPTIONAL_PORTS_AND_SERVICE_DISCOVERY
+    );
+
+    if (!hasPlaceIdSet && isTooComplexStructure(this.props.fields)) {
```

Comment:
In principle you are right. That would be the most generic way.
But for now I wouldn't suggest it.
`isTooComplexStructure` is hard coded in the UI, not connected to the PluginDispatcher.
Bringing this logic to the plugin interface is possible though, but it would be a very complex undertaking. It would be necessary to extend stores, actions, defining new events and introduce some smart logic to signal these states through the dispatcher.